src/main/java/mServer/crawler/sender/MediathekArd.java

```diff
@@ -512,4 +649,51 @@ private String beschreibung(String strUrlFeed) {
 
     }
 
+    private boolean isAllTextsEmpty(String... aTexts)
+    {
+        Boolean isAllTextsEmpty = null;
+        for (String text : aTexts)
+        {
+            isAllTextsEmpty = (isAllTextsEmpty == null || isAllTextsEmpty) && StringUtils.isEmpty(text);
+        }
+        return isAllTextsEmpty == null || isAllTextsEmpty;
+    }
+
+    /**
+     * Searches the Seite for a quality auto to get a M3U8 URL. If the URL is from WRD it searches for the URLs of the MP4 files.
+     *
+     * @param aSeiteStringExtractor The Seite.
+     * @return A Map containing the URLs and Qualities which was found. An empty Map if nothing was found.
+     */
+    private Map<Qualities, String> searchForUrlsWithM3U8(final MSStringBuilder aSeiteStringExtractor)
+    {
+        Map<Qualities, String> urls = new HashMap<>();
+
+        ArrayList<String> patternMatches = new ArrayList<>();
+        aSeiteStringExtractor.extractList(M3U8_PATTERN_START, M3U8_PATTERN_END, patternMatches);
+
+        String m3u8Url = null;
+        for (String patternMatch : patternMatches)
+        {
+            if (patternMatch.startsWith(TEXT_START_HTTP))
+            {
+                m3u8Url = patternMatch;
+                break;
+            }
+        }
+
+        if (m3u8Url != null)
+        {
+            m3u8Url.replaceAll(URL_GET_PARAMETER, "");
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') The return value of "replaceAll" must be used. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2201)